packages/grafana-ui/src/components/uPlot/plugins/ZoomPlugin.tsx

```diff
@@ -50,5 +53,21 @@ export const ZoomPlugin = ({ onZoom, config }: ZoomPluginProps) => {
     });
   }, [config]);
 
+  const refTimeRange = useRef<TimeRange>(timeRange);
+  useEffect(() => {
+    refTimeRange.current = timeRange;
+  }, [timeRange]);
+
+  useLayoutEffect(() => {
+    config.addHook('setCursor', (u) => {
```

Comment:
I misunderstood your comment on this part.

I tried this

```ts
  useLayoutEffect(() => {
    config.setCursor(
      {bind: {
        dblclick: (self, targ, handler)=> {
          const range = refTimeRange.current;
          const diff = range.to.diff(range.from) / 2;
  
          onZoom({ from: range.from.unix() * 1000 - diff, to: range.to.unix() * 1000 + diff });
          
          return handler;
        }
      }}
    );
  }, [config]);
```

but that executes the onZoom infinitely. Maybe I don't understand the api.

Reply:
try returning `null`.

`handler` is uPlot's native dblclick/zoom-out handler so we should ignore it.

Reply:
It still loops `onZoom` call

Reply:
doh, i missed an extra layer of indirection. it should be this:

```js
  useLayoutEffect(() => {
    config.setCursor({
      bind: {
        dblclick: (u, targ, handler) => (e: MouseEvent) => {
          const frTs = refTimeRange.current.from.valueOf();
          const toTs = refTimeRange.current.to.valueOf();
          const pad = (toTs - frTs) / 2;

          onZoom({ from: frTs - pad, to: toTs + pad });

          return null;
        },
      },
    });
  }, [config]);
```
speaking of which, can you just put this into the existing `useLayoutEffect` with the `[config]` dependency instead of creating an additional `useLayoutEffect`?

Reply:
Makes sense, done.